spring-webmvc/src/main/java/org/springframework/web/servlet/mvc/method/annotation/RequestMappingHandlerAdapter.java

```diff
@@ -524,8 +524,11 @@ public void setParameterNameDiscoverer(ParameterNameDiscoverer parameterNameDisc
 	 * in method argument default values.
 	 */
 	@Override
-	public void setBeanFactory(BeanFactory beanFactory) {
-		if (beanFactory instanceof ConfigurableBeanFactory) {
+	public void setBeanFactory(@Nullable BeanFactory beanFactory) {
```

Comment:
Since `getBeanFactory` is protected and returns a narrowed type anyway, the setter's nullability doesn't have to be aligned here, so I'd rather keep the straightforward check.